gensound/transforms.py

```diff
@@ -187,34 +187,49 @@ def realise(self, audio):
 
 ######### Level/ampltidue Stuff ###################
 
-class Fade(Transform):
-    min_fade = -50
-    
-    def __init__(self, is_in=True, duration=3e3):
-        self.is_in = is_in
+class Fade(Transform):   
+    def __init__(self, duration=None, curve=None, is_in=True, degree=1):
+        assert curve in ("linear", "polynomial")
+        # default setting for backwards compatibility (allowing usage of both Fade() and FadeIn/FadeOut()
+        if curve == None: curve = "linear"
+        if duration == None: duration = 3e3
+
+        self.curve = curve
         self.duration = duration
-    
+        self.is_in = is_in
+        self.degree = degree
```

Comment:
as above, the ordering of these lines somewhat makes it appear as if `is_in` is less important than the rest.
Think about it like that: it's most important to know whether it's a fade in or a fade out; next it's important to know the duration; the different curves are minor details, and `degree` should appear next to it since they work in conjunction.